hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1104,6 +1106,7 @@ private class Emit(
         val elt = coerce[TArray](a.typ).elementType
         val accumTypeInfo = coerce[Any](typeToTypeInfo(zero.typ))
         val elementTypeInfoA = coerce[Any](typeToTypeInfo(elt))
+        val xmbody = mb.newField[Boolean]()
```

Comment:
why are the fields named in fold and not named in scan?

Reply:
I've mostly been naming them as needed if something needs to be debugged. The names are otherwise pretty invisible and useless unless there's a bytecode generation error.